lib/AST/Module.cpp

```diff
@@ -835,6 +835,20 @@ void ModuleDecl::getPrecedenceGroups(
   FORWARD(getPrecedenceGroups, (results));
 }
 
+bool ModuleDecl::classifyWitnessAsThrows(SubstitutionMap substitutions) {
```

Comment:
Does this need to be a method on ModuleDecl? Perhaps it should just be a top-level function in the swift namespace defined in SubstitutionMap.cpp

Reply:
no, it just needs to take a `ModuleDecl *` and a `SubstitutionMap` and return a `bool`. It is used in a few spots however. I was unsure of where this should belong since it is not really a part of a module declaration or part of a substitution mapping but instead a thing that applies to the combination of both of those.